### DISCUSSION TITLE
Add additional properties to the meta table

### DISCUSSION THREAD

**Discussion by HMEUW** [ArtesiaWater/hydropandas#97] [Q&A]
I would like to add addtional properties to the meta table of a GroundwaterObs. I my case it is the location of a well along the shoreline; and perpendicular to the shore line. I cannot find this option in the documentation.

Currently I have a workaround to join these properties to an obsCollection.

**Comment by OnnoEbbens** [comment]
It is possible to add extra information to a GroundwaterObs object (or any other ..Obs obejct) by using the `meta` attribute. The `meta` attribute should be a dictionary with additional information. 

When you create an ObsCollection from a list of GroundwaterObs objects you won't see the `meta` information in the ObsCollection DataFrame. If you want to add this information to the DataFrame you can use the `add_meta_to_df` method.

Below a code snippet to make things clear.

```
# generate measurements
ts = pd.DataFrame({'measurements':np.random.randint(0,10,5)}, index=pd.date_range('2020-1-1', '2020-1-5'))

# create Obs object
o1 = hpd.Obs(ts, name='obs',x=0, y=0, 
             meta={'dijkpaal': 'dp241', 'afstand_tot_dijk': 342})

# show metadata
print(o1.meta)

# create ObsCollection
oc = hpd.ObsCollection.from_list([o1])

# add metadata to an ObsCollection
oc.add_meta_to_df('dijkpaal')
oc.add_meta_to_df('afstand_tot_dijk')

print(oc)
```

**Comment by HMEUW** [comment]
Please mind that the behaviour of `add_met_to_df` has changed in HydroPandas updates since Onno's post. Current code is:

```
# add metadata as column to the ObsCollection
oc = oc.add_meta_to_df('dijkpaal')
oc = oc.add_meta_to_df('afstand_tot_dijk')
```
